tests/queries/0_stateless/02477_age.sql

```diff
@@ -0,0 +1,82 @@
+SELECT 'Various intervals';
+
+SELECT age('year', toDate('2017-12-31'), toDate('2016-01-01'));
+SELECT age('year', toDate('2017-12-31'), toDate('2017-01-01'));
```

Comment:
The "negative" calculations (1st argument > 2nd argument) are a bit weird. E.g. is it really expected that the year component of the absolute difference between 01-01 and 12-31 is -1 years? It is more like 0.99999 years = 0 years.

Reply:
I corrected it and pushed my changes. Please take a look.

To compare startdate > enddate we cannot rely on `res` like:
`if (res < 0)`
because in some cases res ==0 for both for startdate > enddate and for startdate < enddate

`res` is calculated as here:
```
 auto res = static_cast<Int64>(transform_y.execute(y, timezone_y))
                - static_cast<Int64>(transform_x.execute(x, timezone_x));
```
So I had to calculate seconds first and then compare the seconds. 
```
            auto x_seconds = TransformDateTime64<ToRelativeSecondNumImpl<ResultPrecision::Extended>>(transform_x.getScaleMultiplier()).execute(x, timezone_x);
            auto y_seconds = TransformDateTime64<ToRelativeSecondNumImpl<ResultPrecision::Extended>>(transform_y.getScaleMultiplier()).execute(y, timezone_y);
            if (x_seconds <= y_seconds)

```
Because as far as I know there are no  DateTimeTransforms to compare  different types of x and y. For example when `x` is `Date32` and `y` is `DateTime64`.

I also renamed `x_comp` to `a_comp` and `y_comp` to `b_comp` to avoid confusion what is `x` or `y` in later comparisons.

Reply:
Looks good to me, thanks!

Reply:
Great. Thank you for reviewing it!